ddtrace/contrib/trace_utils.py

```diff
@@ -240,12 +240,8 @@ def get_header_value(key):  # type: (str) -> Optional[str]
     if peer_ip:
         try:
             peer_ip_obj = ipaddress.ip_address(six.text_type(peer_ip))
-            if not peer_ip_obj.is_loopback:
-                # If we have a private_ip and the peer_ip is not public, prefer the one we have
-                if private_ip:
-                    if peer_ip_obj.is_loopback or peer_ip_obj.is_private:
-                        return private_ip
-                # peer_ip is public or we dont have a private_ip to we return the private peer_ip
+            # peer_ip is public or we dont have a private_ip to we return the private peer_ip
+            if not (peer_ip_obj.is_private or peer_ip_obj.is_loopback):
```

Comment:
I think it misses the case when `private_ip` is empty, in this case we should always return the peer ip.

Reply:
Not sure it was a problem (the `if tmp_ip_header_value` should avoid empty strings) but I refactored the code a little and added some tests for peer selection including the headers ip being empty, PTAL.